extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java

```diff
@@ -87,9 +101,27 @@ void build(BuildProducer<FeatureBuildItem> featureProducer,
         List<String> applicationMigrations = discoverApplicationMigrations(getMigrationLocations(dataSourceNames));
         recorder.setApplicationMigrationFiles(applicationMigrations);
 
+        Set<Class<?>> javaMigrationClasses = new HashSet<>();
+        addJavaMigrations(combinedIndexBuildItem.getIndex().getAllKnownImplementors(JAVA_MIGRATION), context,
+                reflectiveClassProducer, javaMigrationClasses);
+        addJavaMigrations(combinedIndexBuildItem.getIndex().getAllKnownSubclasses(BASE_JAVA_MIGRATION), context,
```

Comment:
`BaseJavaMigration implements JavaMigration`, wouldn't classes from the statement above be included here?

Reply:
I wonder if we shouldn't just stick to the first statement

Reply:
Unfortunately no - I tried it (unless I tried wrongi)

Reply:
Perhaps the Flyway JAR need to be Jandexed too?

Reply:
Done :wink: See if that makes sense

Reply:
If the tests pass, I'm fine with it, thanks!